tests/e2e-pw/tests/product-gallery/inner-blocks/product-gallery-thumbnails/product-gallery-thumbnails.block_theme.spec.ts

```diff
@@ -0,0 +1,374 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+
+/**
+ * Internal dependencies
+ */
+import { addBlock } from './utils';
+
+const blockData = {
+	name: 'woocommerce/product-gallery-thumbnails',
+	mainClass: '.wp-block-woocommerce-product-gallery-thumbnails',
+	selectors: {
+		frontend: {},
+		editor: {
+			thumbnails: '.wp-block-woocommerce-product-gallery-thumbnails',
+			noThumbnailsOption: 'button[data-value=off]',
+			leftPositionThumbnailsOption: 'button[data-value=left]',
+			bottomPositionThumbnailsOption: 'button[data-value=bottom]',
+			rightPositionThumbnailsOption: 'button[data-value=right]',
+		},
+	},
+	slug: 'single-product',
+	productPage: '/product/v-neck-t-shirt/',
+};
+
+test.describe( `${ blockData.name }`, () => {
+	test.beforeEach( async ( { requestUtils, admin, editorUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await admin.visitSiteEditor( {
+			postId: `woocommerce/woocommerce//${ blockData.slug }`,
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test( 'Renders Product Gallery Thumbnails block on the editor side', async ( {
+		editorUtils,
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'woocommerce/product-gallery',
+		} );
+
+		const block = await editorUtils.getBlockByName( blockData.name );
+
+		await expect( block ).toBeVisible();
+	} );
+
+	test( 'Renders Product Gallery Thumbnails block on the frontend side', async ( {
```

Comment:
This test is identical to `'Renders Product Gallery Thumbnails block on the editor side'`, so I recommend updating it to test the render on the FE truly.

What do you think about combining `'Renders Product Gallery Thumbnails block on the editor side'` and `'Renders Product Gallery Thumbnails block on the frontend side'` in a single test? This way we can insert the `'woocommerce/product-gallery'` block just once and expedite the tests.